### DISCUSSION TITLE
Transelate the position of a model in AR mode

### DISCUSSION THREAD

**Discussion by AntonioMartinezC** [google/model-viewer#4302] [Q&A] [answered]
Heho, 

So I want to be able to reposition my model in AR mode in relation to the anchor on the ground. More specificly I need the model to go vertically up and vertically down with some sort of control over it (touch interaction might be to much as rotating, scaling and positioning is already bound to those) 

So I want to have a slider that makes my model float higher or lower depending on the sliders value. Is that possible with modle viewer? I sadly can't find a lot of info on that

**Comment by danilokrulj** [comment]
It is possible (or atleast was) but only in WebXR mode, but on iOS you already have that logic built in, just drag with two fingers up/down and the model will change position vertically. For WebXR I used this code (its from some old project so im not sure if it works, but maybe it will point you in some direction): 

Good luck :)

`     var ts,tp;

      function handleStart(evt) {
        evt.preventDefault();
        switch (evt.targetTouches.length) {
        case 2:
          // Two simultaneous touches
          ts = evt.touches[0].clientY;
          break;
        default:
          // More than two simultaneous touches
          //evt.target.style.background = "lightblue";
        }
      }

      let moveStep = 0.05;

      function handleMove(evt) {
        evt.preventDefault();
        switch (evt.targetTouches.length) {
        case 2:
          // Two simultaneous touches
          var te = evt.changedTouches[0].clientY;
          
          if(te<tp-5){ // 5 je broj predjenih piksela
            moveableObj.position.y += moveStep;
          }else if(te>tp+5){
            if(moveableObj.position.y > 0)
            moveableObj.position.y -= moveStep;
          }
          tp = te;
          if(moveableObj.position.y < 1){
            modelViewer.shadowIntensity = 2;
            mw[sceneSym].children[0].children[0].children[0].children[1].children[1].visible = false;
          }else{
            modelViewer.shadowIntensity = 0;
            mw[sceneSym].children[0].children[0].children[0].children[1].children[1].visible = true;
          }
          break;
        default:
          // More than two simultaneous touches
          //evt.target.style.background = "lightblue";
        }
      }`

**Reply by elalish** [reply, marked as answer]
That's well outside of our public API. We have been considering some new APIs for this kind of thing though. Meanwhile, another option would be to export the file with an animation that moves it up, then select the height with `currentTime`.

**Reply by AntonioMartinezC** [reply]
Thought about the animation workaraund aswell but it feels like that would produce a lot of room for problems down the line as I Ideally want to use three.js's gltf exporter to dynamically change the file and would have to adjust the animation which would not be impossible but just another thing to consider I guess ^^

**Reply by AntonioMartinezC** [reply]
Thanks for the answers btw!
